### PR TITLE
Document UI Library Options

### LINKED ISSUE #3750: Document options for UI libraries
This is a common question, we should answer it in docs. We can recommend react-native-paper and react-native-elements, there may be others but I haven't kept up.

### PR DESCRIPTION
**Pull request by cruzach**
# Why

It's a common question we get, and this will provide support to developers of all levels. 
And it closes [#3750](https://github.com/expo/expo/issues/3750)

# How

Looked up different library options and made the .md file

# Test Plan

Look at the .md file

### CODE REVIEW

**Comment by antgel** [comment]
Looks like two commits are duplicated. Suggest dropping one.

**Comment by cruzach** [comment]
They're on different versions: first commit is to unversioned, second to v32, third to v31

**Comment by antgel** [comment]
My bad. :)

File: docs/pages/versions/unversioned/guides/userinterface.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: User Interface
+---
+
+The User Interface (UI) is a huge part of any app. If the icon is your front door, the UI is the interior design. You can definitely take the route of completely designing your own UI from scratch, but if you want to get it up and running fast while still having a polished finish, there are plenty of libraries already in place to help get your app where you want it.
+
+Below (in no particular order) are some of the most popular UI Libraries. Test them all out and see what you like!
+
+* [React Native Elements](https://react-native-training.github.io/react-native-elements/), and their [Docs](https://react-native-training.github.io/react-native-elements/docs/getting_started.html)
```
Comment (brentvatne) [inline docs/pages/versions/unversioned/guides/userinterface.md]: we should add React Native Paper to the top of this list:

https://github.com/callstack/react-native-paper / https://callstack.github.io/react-native-paper/

**Review by brentvatne — CHANGES_REQUESTED** [review]

File: docs/pages/versions/unversioned/guides/userinterface.md (hunk 1 above)
Comment (cruzach) [reply docs/pages/versions/unversioned/guides/userinterface.md]: Added 👍

File: docs/pages/versions/v31.0.0/guides/userinterface.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+title: User Interface
+---
+
+The User Interface (UI) is a huge part of any app. If the icon is your front door, the UI is the interior design. You can definitely take the route of completely designing your own UI from scratch, but if you want to get it up and running fast while still having a polished finish, there are plenty of libraries already in place to help get your app where you want it.
+
+Below (in no particular order) are some of the most popular UI Libraries. Test them all out and see what you like!
+
+* [React Native Paper](https://github.com/callstack/react-native-paper), and their [Docs](https://callstack.github.io/react-native-paper/index.html)
```
Comment (brentvatne) [inline docs/pages/versions/v31.0.0/guides/userinterface.md]: looks like we're still missing React Native UI Kit for iOS - https://github.com/callstack/react-native-ios-kit / https://callstack.github.io/react-native-ios-kit/

once that's added we need to make sure the docs are the same across all sdk versions, looks like there's a bit of a difference right now

Comment (cruzach) [reply docs/pages/versions/v31.0.0/guides/userinterface.md]: Was going to add this one, but realized last commit was September of 2018. That's ~6 months, you think still just go ahead and add anyways?

Comment (brentvatne) [reply docs/pages/versions/v31.0.0/guides/userinterface.md]: I think it works pretty well still and I suspect callstack will remain responsive to issues on it, cc @satya164

Comment (satya164) [reply docs/pages/versions/v31.0.0/guides/userinterface.md]: Unfortunately, it's not actively maintained by anyone atm coz there's not much interest in it (only 13 downloads per week on npm)

Comment (brentvatne) [reply docs/pages/versions/v31.0.0/guides/userinterface.md]: 😢

**Review by brentvatne — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

**Pull request merged**